DiceKit/MaximizationExpression.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  MaximizationExpression.swift
+//  DiceKit
+//
+//  Created by Logan Johnson on 8/15/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import Foundation
+
+public struct MaximizationExpression<Expression: protocol<ExpressionType, Equatable>>: Equatable {
+    
+    public let expression: Expression
+    
+    public init(_ expression: Expression) {
+        self.expression = expression
+    }
+    
+}
+
+// MARK: - ExpressionType
+
+extension MaximizationExpression: ExpressionType {
+    
+    public typealias Result = MaximizationExpressionResult<Expression.Result>
+    
+    public func evaluate() -> Result {
+        if let maximumOutcome = expression.probabilityMass.maximumOutcome() {
```

Comment:
Let's try to follow the golden path rule. Make this a `guard` instead and early return with the `0`.